p3/plot/backend/pgfplots.py

```diff
@@ -0,0 +1,434 @@
+# Copyright (c) 2022-2023 Intel Corporation
+# SPDX-License-Identifier: MIT
+"""
+Contains objects for interacting with plots produced using the
+:py:mod:`pgfplots` backend.
+"""
+
+import os
+import string
+
+import jinja2
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+import p3.metrics
+from p3._utils import _require_numeric
+from p3.plot._common import ApplicationStyle, Legend, PlatformStyle
+from p3.plot.backend import CascadePlot, NavChart
+
+
+class CascadePlot(CascadePlot):
+    """
+    Cascade plot object for :py:mod:`pgfplots`.
+    """
+
+    def __get_colors(self, applications, kwarg):
+        """
+        Assign a color to each application based on supplied kwarg.
+        """
+        if isinstance(kwarg, str):
+            cmap = getattr(plt.cm, kwarg)
+        elif isinstance(kwarg, list):
+            cmap = matplotlib.colors.ListedColormap(kwarg)
+        elif isinstance(kwarg, matplotlib.colors.Colormap):
+            cmap = kwarg
+        else:
+            raise ValueError("Unsupported type provided for colormap.")
+
+        cmap = cmap.resampled(len(applications))
+        colors = cmap(np.linspace(0, 1, len(applications)))
+        return {app: color for app, color in zip(applications, colors)}
```

Comment:
Certainly the function shouldn't be defined twice in each file (in CascadePlot and NavChart). But I would say that each should probably have its own implementation in there so that there isn't a strange dependence.

Ideally I would remove the reliance on matplotlib from the pgfplots backend at some point in the future, but it's certainly an easier way to do with getting appropriate colours at this time.

I'll pull the function up in each file so that there's only a single implementation in each backend rather than 2.